### PR TITLE
refs #955: deprecate Client::AUTH_* constants and replace them with AuthMethod::AUTH_* const

### PR DESCRIPTION
**Pull request by ipalo**
Contribution for #955

### CODE REVIEW

File: lib/Github/AuthMethod.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+namespace Github;
+
+final class AuthMethod
+{
+    /**
+     * Authenticate using a client_id/client_secret combination.
+     *
+     * @var string
+     */
+    const AUTH_CLIENT_ID = 'client_id_header';
```
Comment (acrobat) [inline lib/Github/AuthMethod.php:12]: I woud remove the `AUTH_` prefix for these constants as the classname gives that context already (same for the other constants)

File: lib/Github/Client.php [hunk 2]
```diff
@@ -73,13 +73,17 @@ class Client
      * Authenticate using a client_id/client_secret combination.
      *
      * @var string
+     *
+     * @deprecated Use the AuthMethod const
```
Comment (acrobat) [inline lib/Github/Client.php:77]:
```suggestion
     * @deprecated Use the Github\AuthMethod::CLIENT_ID const
```

Directly suggest the corresponding constant to make it easier for users who need to change their code

File: UPGRADE-3.0.md [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 ### Authentication methods
 
-* `Github\Client::AUTH_URL_TOKEN` use `Github\Client::AUTH_ACCESS_TOKEN` instead.
-* `Github\Client::AUTH_URL_CLIENT_ID` use `Github\Client::AUTH_CLIENT_ID` instead.
-* `Github\Client::AUTH_HTTP_TOKEN` use `Github\Client::AUTH_ACCESS_TOKEN` instead.
-* `Github\Client::AUTH_HTTP_PASSWORD` use `Github\Client::AUTH_ACCESS_TOKEN` instead.
+* `Github\Client::AUTH_ACCESS_TOKEN`  use `Github\AuthMethod::AUTH_ACCESS_TOKEN` instead.
```
Comment (acrobat) [inline UPGRADE-3.0.md:18]: Please revert these changed as this is the upgrade file for v2 -> v3. Add your upgrade notes to the `UPGRADE-4.0.md` file.

**Review by acrobat — CHANGES_REQUESTED** [review]
Hi @ipalo, thanks for the PR! I've left a few comments on things that should be updated!

File: lib/Github/AuthMethod.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+namespace Github;
+
+final class AuthMethod
+{
+    /**
+     * Authenticate using a client_id/client_secret combination.
+     *
+     * @var string
+     */
+    const CLIENT_ID = 'client_id_header';
```
Comment (acrobat) [inline lib/Github/AuthMethod.php:12]: Add the `public` const visibility to the constants in this class.

```suggestion
    public const CLIENT_ID = 'client_id_header';
```

**Review by acrobat — CHANGES_REQUESTED** [review]
One last minor change, otherwise the PR is good to go!

**Review by acrobat — APPROVED** [review]

**Comment by acrobat** [comment]
Thanks @ipalo! And congrats on your first contribution! 🎉

**Comment by ipalo** [comment]
> Thanks @ipalo! And congrats on your first contribution! 🎉

Hi @acrobat! I noticed the PR hasn't yet been accepted for the Hacktoberfest. Could you unblock it with the "accepted" label? Thanks 🙌

**Comment by acrobat** [comment]
@ipalo this should be ok already as the repository has the "hacktoberfest" topic. But I've added the accepted label to be sure.

**Pull request merged**